sql/core/src/main/scala/org/apache/spark/sql/execution/command/functions.scala

```diff
@@ -157,6 +157,9 @@ case class DropFunction(
         throw new AnalysisException(s"Specifying a database in DROP TEMPORARY FUNCTION " +
           s"is not allowed: '${databaseName.get}'")
       }
+      if (FunctionRegistry.builtin.functionExists(functionName)) {
```

Comment:
I also think what you said is valid. If so, I think we also need to provide users a way to recover the built-in function, if they dropped it for any purpose. 

This PR is just to make our behavior consistent with Hive and the mainstream RDBMS. Normally, we do not allow users to drop the built-in functions. I am fine if we allow users to drop the built-in functions. 

Thanks!